test/unit/active_record_batch_enumerator_test.rb

```diff
@@ -48,6 +48,13 @@ class ActiveRecordBatchEnumeratorTest < IterationUnitTest
       assert_equal 4, enum.size # 3 batches of 3, 1 batch of 1
     end
 
+    test "#size returns size of the Enumerator with a subset of columns" do
+      enum = build_enumerator(relation: Product.select(:id, :name))
+      assert_equal 5, enum.size # 5 batches of 2
+      enum = build_enumerator(relation: Product.select(:id, :name), batch_size: 3)
```

Comment:
Also `select` for batches is kind of weird because you shouldn't operate on the records, just on the batches (`update_all`/`delete_all`/`destroy_all`) so you shouldn't need to list columns. But we should still fix it 👍

Reply:
Yeah, that's a good point. The issue actually surfaced when using batches, and using an `update_all` call, but using `select` for only the columns we needed as a way to (micro?) optimize the job to really only fetch what we needed from the DB. Probably only realistically saved not that much memory in that one case, but for larger batches, it might.